### PR TITLE
Adds the sdedit recipe.

### PR DESCRIPTION
**Pull request by dbsr**

### CODE REVIEW

File: Library/Formula/sdedit.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  sha256 "86f52193ceb3a478fd0cb356f631b6131bc649561fcc5effe73edfb3b12c33c9"
+  depends_on :java
+
+  def install
+    libexec.install "sdedit-#{version}.jar"
+    bin.write_jar_script libexec/"sdedit-#{version}.jar", "sdedit"
+  end
+
+  test do
+    system "#{bin}/sdedit", "--help"
```
Comment (mikemcquaid) [inline Library/Formula/sdedit.rb]: Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!

File: Library/Formula/sdedit.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class Sdedit < Formula
+  desc "Tool for generating sequence diagrams very quickly."
+  homepage "http://sdedit.sourceforge.net"
+  url "https://downloads.sourceforge.net/projects/sdedit/files/sdedit/4.0/sdedit-4.01.jar"
+  sha256 "86f52193ceb3a478fd0cb356f631b6131bc649561fcc5effe73edfb3b12c33c9"
```
Comment (mikemcquaid) [inline Library/Formula/sdedit.rb]: This should be: `49db33ac79775608bcba439eb53a84d59d6906d6539528f6d58a4e8601f741c5`

**Comment by dbsr** [comment]
I modified the test but still cant the SHA256 check to pass. It seems every time I download / create the sdedit recipe it has a difference SHA256 :8ball:

**Comment by dunn** [comment]
Looks like the SHA matches now.  Just a small audit failure:
```
== /usr/local/Library/Formula/sdedit.rb ==
C: 19: 97: Unnecessary spacing detected.

1 file inspected, 1 offense detected
```

File: Library/Formula/sdedit.rb [hunk 3]
```diff
@@ -0,0 +1,21 @@
+class Sdedit < Formula
+  desc "Tool for generating sequence diagrams very quickly."
+  homepage "http://sdedit.sourceforge.net"
+  url "https://downloads.sourceforge.net/project/sdedit/sdedit/4.0/sdedit-4.01.jar"
+  sha256 "060576f9fe79bda0a65f2cfa0b041fceaf7846f034a7519ef939b73ae82673f1"
+  depends_on :java
```
Comment (DomT4) [inline Library/Formula/sdedit.rb]: Please leave an empty line between the `sha256` and the `depends_on`, Thanks!

File: Library/Formula/sdedit.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+    bin.write_jar_script libexec/"sdedit-#{version}.jar", "sdedit"
+  end
+
+  test do
+    (testpath/"test.sd").write <<-EOS.undent
+    #![SD ticket order]
+    ext:External[pe]
+    user:Actor
+    EOS
+    system "java", "-jar", "#{libexec}/sdedit-4.01.jar", "-t", "pdf", "-o", testpath/"test.pdf",  testpath/"test.sd"
```
Comment (DomT4) [inline Library/Formula/sdedit.rb]: Please break this onto two lines so it doesn't stretch the formula so much.

**Comment by dbsr** [comment]
squashed commits.

**Comment by dbsr** [comment]
Can you please re-review this PR?

File: Library/Formula/sdedit.rb [hunk 5]
```diff
@@ -0,0 +1,23 @@
+class Sdedit < Formula
+  desc "Tool for generating sequence diagrams very quickly."
+  homepage "http://sdedit.sourceforge.net"
+  url "https://downloads.sourceforge.net/project/sdedit/sdedit/4.0/sdedit-4.01.jar"
+  sha256 "060576f9fe79bda0a65f2cfa0b041fceaf7846f034a7519ef939b73ae82673f1"
+
+  depends_on :java
```
Comment (DomT4) [inline Library/Formula/sdedit.rb]: Does it require any specific version of Java?

**Comment by dbsr** [comment]
Squashed commits again. Thanks for the help and feedback.

**Comment by dbsr** [comment]
Please re-review :)

**Comment by DomT4** [comment]
Test is only failing here because Sourceforge have taken a bunch of homepages offline to deal with last night's outage.

Merged in https://github.com/Homebrew/homebrew/commit/f9510c5acce9ffe53594425687547ed29fe84b41. Thank you for your contribution to Homebrew @dbsr; we appreciate it! :smiley_cat:.

Just for future reference, ideal commit message for new formula is styled:
```
sdedit 4.01 (new formula)
```
Thanks!

**Pull request closed without merging**